packages/adapter-flagsmith/src/index.ts

```
@@ -0,0 +1,54 @@
+import type { Adapter } from 'flags';
+import flagsmith, { IFlagsmithFeature, IInitConfig } from 'flagsmith';
+
+type AdapterParams = {
+  key?: string;
+} & IInitConfig;
+
+export function createFlagsmithAdapter<
+  ValueType extends IFlagsmithFeature,
+  EntitiesType,
+>({
+  key: customKey,
+  ...configParams
+}: AdapterParams): Adapter<ValueType, EntitiesType> {
+  async function initialize(config: IInitConfig) {
+    if (flagsmith?.initialised) return;
+    await flagsmith.init({ fetch: global.fetch, ...configParams });
+  }
```

It looks like flagsmith may be initialized with a user — In that case, this pattern won't work because we'll never re-initialize for another user. I'll need to better understand how `identity` fits in here, and how it will handle the case where there's multiple different users depending on the request.

https://github.com/Flagsmith/flagsmith-js-examples/blob/main/nextjs-middleware/middleware.ts#L15

---

@AAorris unfortunately for now we don't support multiple different users. We can throw an error in case of multiple users in a request.

---

The issue here seems to be that the adapter will initialize once and then flags may be called many times with different requests (Ex. if the function is warm, especially using [fluid compute](https://vercel.com/fluid)

```tsx
async function identify({ cookies }) {
  return cookies.get('user')
}

export const exampleFlag = flag({
  key: 'exampleFlag',
  adapter: flagSmithAdapter.getFeature(),
  identify
})
```
above: illustrating adapter usage with dynamic context

It would be best to initialize the adapter in such a way that it only fetches Flagsmith rules/logic, and then evaluates with user info during the `decide()` function